File: client/src/data/models/DeliveryTask.ts
import {DeliveryTaskStatus} from "@data/models/DeliveryTaskStatus";
import {Order} from "@data/models/Order";
import { Type } from "class-transformer";
import {User} from "@data/models/User";

export class DeliveryTask {
    id!: number;

    @Type(() => User)
    user!: User;

    @Type(() => DeliveryTaskStatus)
    status!: DeliveryTaskStatus;

    @Type(() => Order)
    orders!: Order[];
}

File: backend/data/model/deliverytask.py
from datetime import datetime
from typing import Optional, List

from sqlmodel import SQLModel, Field, Relationship

from data.model.deliverytaskstatus import DeliveryTaskStatus


class DeliveryTask(SQLModel, table=True):
    __tablename__ = "deliverytask"

    id: Optional[int] = Field(default=None, primary_key=True)
    user_id: int = Field(foreign_key="user.id", nullable=True)
    date: datetime = Field(default=None, nullable=False)
    user: Optional["User"] = Relationship(
        back_populates="delivery_tasks",
    )
    status_id: int = Field(foreign_key="deliverytaskstatus.id")
    status: Optional[DeliveryTaskStatus] = Relationship(
        back_populates="delivery_tasks"
    )
    orders: List["Order"] = Relationship(
        back_populates="delivery_task"
    )
    delivery_way_len: float = Field(default=0)
